pennylane/resources/factorization.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for two-electron tensor factorization.
+"""
+from pennylane import numpy as np
+
+
+def factorize(two, tol_first, tol_second):
```

Comment:
Two questions:

- Is it worth having default tolerance values?
- From reading the docstring below, I can't work out how `tol_first` and `tol_second` differ; as far as I can tell, one is for factor tolerance, the other is for eigenvalue tolerance? Would `tol_factor` and `tol_eigval` be better names?

Reply:
Both points are correct and added.